framework-test/src/main/java/org/checkerframework/framework/test/diagnostics/TestDiagnosticUtils.java

```diff
@@ -22,13 +22,23 @@ public class TestDiagnosticUtils {
 
     /** How the diagnostics appear in Java source files. */
     public static final String DIAGNOSTIC_IN_JAVA_REGEX =
-            "\\s*(error|fixable-error|warning|fixable-warning|other):\\s*(\\(?.*\\)?)\\s*";
+            "\\s*(error|fixable-error|warning|fixable-warning|other):\\s*(\\(?[\\s\\S]*\\)?)\\s*";
 
     /** How the diagnostics appear in Java source files. */
+    // Enables whitespace check, so it detects line separator \r\n in Windows.
     public static final Pattern DIAGNOSTIC_IN_JAVA_PATTERN =
             Pattern.compile(DIAGNOSTIC_IN_JAVA_REGEX);
 
-    public static final String DIAGNOSTIC_WARNING_IN_JAVA_REGEX = "\\s*warning:\\s*(.*\\s*.*)\\s*";
+    /**
+     * Regular expression pattern to match Java diagnostic warnings.
+     *
+     * <p>Include white space check, which is necessary for detecting multi-line warning messages on
+     * both linux and Windows operating system.
+     */
+    public static final String DIAGNOSTIC_WARNING_IN_JAVA_REGEX =
+            "\\s*warning:\\s*([\\s\\S]*\\s*[\\s\\S]*)\\s*";
```

Comment:
I don't know what the original motivation for this pattern was... wouldn't `[\\s\\S]*` be simpler and allow the same things? What's the point of `\\s*` in the middle?